spec/negotiation.html

```diff
@@ -207,6 +207,11 @@ <h1>ResolveLocale ( _availableLocales_, _requestedLocales_, _options_, _relevant
           1. If _options_ has a field [[&lt;_key_&gt;]], then
             1. Let _optionsValue_ be _options_.[[&lt;_key_&gt;]].
             1. Assert: Type(_optionsValue_) is either String, Undefined, or Null.
+            1. If Type(_optionsValue_) is String, then
+              1. Let _optionsValue_ be the string _optionsValue_ after performing the algorithm steps to transform Unicode extension values to canonical syntax per <a href="https://unicode.org/reports/tr35/#Canonical_Unicode_Locale_Identifiers">Unicode Technical Standard #35 LDML § 3.2.1 Canonical Unicode Locale Identifiers</a>.
+              1. Let _optionsValue_ be the string _optionsValue_ after performing the algorithm steps to replace Unicode extension values with their canonical form per <a href="https://unicode.org/reports/tr35/#Canonical_Unicode_Locale_Identifiers">Unicode Technical Standard #35 LDML § 3.2.1 Canonical Unicode Locale Identifiers</a>.
```

Comment:
This seems like it needs to be done with respect to _key_, after all "ca-islamicc" should be transformed but "nu-islamicc" shouldn't.  Maybe something like

```
1. Treating _key_ as a <a href="https://unicode.org/reports/tr35/#unicode_locale_extensions">`key`</a> and _optionsValue_ as its associated <a href="https://unicode.org/reports/tr35/#unicode_locale_extensions">`type`</a> in a Unicode locale extension extension sequence,
  1. Transform _optionsValue_ to have canonical syntax per <a href="https://unicode.org/reports/tr35/#Canonical_Unicode_Locale_Identifiers">Unicode Technical Standard #35 LDML § 3.2.1 Canonical Unicode Locale Identifiers</a>.  (This will convert to lowercase.)
  1. Transform _optionsValue_ to be in canonical form per <a href="https://unicode.org/reports/tr35/#Canonical_Unicode_Locale_Identifiers">Unicode Technical Standard #35 LDML § 3.2.1 Canonical Unicode Locale Identifiers</a>.  (This will replace an alias with its canonical form.)
```

instead of the two steps here?  Best I've got here at this point...

Reply:
Added "treating _key_ as `ukey' and _optionsValue_ as `uvalue`" to both steps, because I didn't want to add a new algorithm step notation.